client/environment.go

```diff
@@ -58,14 +58,29 @@ const (
 	TTlTypeInfinite TTLType = "INFINITE"
 )
 
+type SubEnvironment struct {
+	Revision             string               `json:"revision,omitempty"`
+	Workspace            string               `json:"workspace,omitempty"`
+	ConfigurationChanges ConfigurationChanges `json:"configurationChanges"`
+}
+
 type DeployRequest struct {
-	BlueprintId          string                `json:"blueprintId,omitempty"`
-	BlueprintRevision    string                `json:"blueprintRevision,omitempty"`
-	BlueprintRepository  string                `json:"blueprintRepository,omitempty"`
-	ConfigurationChanges *ConfigurationChanges `json:"configurationChanges,omitempty"`
-	TTL                  *TTL                  `json:"ttl,omitempty"`
-	EnvName              string                `json:"envName,omitempty"`
-	UserRequiresApproval *bool                 `json:"userRequiresApproval,omitempty"`
+	BlueprintId          string                    `json:"blueprintId,omitempty"`
+	BlueprintRevision    string                    `json:"blueprintRevision,omitempty"`
+	BlueprintRepository  string                    `json:"blueprintRepository,omitempty"`
+	ConfigurationChanges *ConfigurationChanges     `json:"configurationChanges,omitempty"`
+	TTL                  *TTL                      `json:"ttl,omitempty"`
+	EnvName              string                    `json:"envName,omitempty"`
+	UserRequiresApproval *bool                     `json:"userRequiresApproval,omitempty"`
+	SubEnvironments      map[string]SubEnvironment `json:"subEnvironments,omitempty"`
+}
+
+type WorkflowEnvironment struct {
+	EnvironmentId string `json:"environmentId"`
+}
+
+type WorkflowFile struct {
+	Environments map[string]WorkflowEnvironment `json:"environments"`
```

Comment:
not sure about the name `WorkflowEnvironment`, the workflow file is a map between the its sub-environment aliases and the sub-environments.